src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -1950,14 +1950,16 @@ Method* SystemDictionary::find_method_handle_intrinsic(vmIntrinsicID iid,
   // Method has been added as the value.
   {
     MonitorLocker ml(THREAD, InvokeMethodIntrinsicTable_lock);
-    while (met == nullptr) {
+    while (true) {
```

Comment:
Given that you break out of this loop based on the state of `created`, it might be better to change the loop to `while(!created)`.

Reply:
I agree, that does look better.  Thanks!

Reply:
No it doesn't! Sorry but there are two reasons for getting out of the loop - one involves a return and one involves the break - only putting one of those in the loop condition doesn't clarify anything IMO.

Reply:
So I have 4 reviewers and 3 different opinions, on something that people's eyes differ on but are the same thing.

Reply:
By that logic, all loops should be while (true) with a break in the middle for the condition?

Reply:
As you know we spent some time trying to express this loop correctly and using the "obvious" loop conditions did not work because we had to re-check the condition after the loop to figure out whether we needed to return or not. The `while(true)` captured the fact there there was no simple/obvious loop termination condition here and that you need to check the loop body to see exactly what conditions exit the loop and in what way. Picking one of those conditions gives a false impression about the operation of the loop IMO and doesn't aid in understandability. YMMV.

Reply:
It was a lot of work to reason about this loop.  I changed it back.